boltrouter/bolt_request.go

```diff
@@ -346,6 +408,47 @@ func (br *BoltRouter) doAwsRequest(logger *zap.Logger, boltReq *BoltRequest, isF
 	return resp, isFailover, err
 }
 
+func (br *BoltRouter) doGcpRequest(logger *zap.Logger, boltReq *BoltRequest, isFailover bool, analytics *BoltRequestAnalytics) (resp *http.Response, isFailoverRequest bool, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			logger.Error("panic occurred during GCP request", zap.Any("panic", r))
+			resp = nil
+			err = ErrPanicDuringAwsRequest
+		}
+	}()
```

Comment:
I don't understand why this is needed, gcp and aws never panics. If we get a panic, we probably want to panic.

The session in API already has a recover middleware.

if this panics, it's a us issue, not a aws or gcp issue. Or are there edge cases aws/gcp panics?

Reply:
The idea is that clients should never see anything that's not a valid http response from Sidekick.

Session in API recovers when proper errors are returned and calls internalerror - it does not handle panics in newboltrequest or dorequest atm